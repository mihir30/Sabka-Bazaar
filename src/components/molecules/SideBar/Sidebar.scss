.category-container {
    background-color: #EAEAEA;

    .category-list {
        list-style-type: none;
        font-size: 1rem;
        padding: 0;
        margin: 0;
        cursor: pointer;

        li:hover {
            background-color: #cc0044;
            color: #fff;
        }

        li {
            border-bottom: 1px solid rgb(177, 173, 173);
            padding: 1rem;
            font-size: 1rem;
        }

        .active {
            background-color: #cc0044;
            color: #fff;
        }
    }
}