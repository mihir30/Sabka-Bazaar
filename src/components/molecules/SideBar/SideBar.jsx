import React from "react";

import "./Sidebar.scss";

const SideBar = ({ categoryData, ...props }) => {
  return (
    <div className="category-container">
      <ul className="category-list">
        {categoryData?.map((category) =>
          category.enabled ? (
            <li
              key={category.id}
              className={category.id === props.id ? "active" : null}
              onClick={() => props.handleCategoryChange(category.id)}
            >
              {category.name}
            </li>
          ) : null
        )}
      </ul>
    </div>
  );
};

export default SideBar;
