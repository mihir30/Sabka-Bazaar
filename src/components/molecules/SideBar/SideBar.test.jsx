import React from 'react';
import renderer from 'react-test-renderer';
import SideBar from './SideBar';

it ("Checking test for SideBar", () => {
    const tree = renderer.create(<SideBar />).toJSON();
    expect(tree).toMatchSnapshot();
});