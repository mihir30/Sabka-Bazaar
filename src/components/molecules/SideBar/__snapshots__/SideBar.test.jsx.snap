// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking test for SideBar 1`] = `
<div
  className="category-container"
>
  <ul
    className="category-list"
  />
</div>
`;
